tstate/tstate_view.go

```diff
@@ -230,13 +239,21 @@ func (ts *TStateView) Insert(ctx context.Context, key []byte, value []byte) erro
 		pastWrites:    chunks(ts.writes, k),
```

Comment:
You should leave a comment on `getValue` that says this is ok to call here because `state.Write` provide exclusive access to state. You may want to comment that as an invariant.